### PR TITLE
Client L2 writing

### PR DESCRIPTION
**Pull request by pikonha**
# Client L2 writing

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

This introduces the handling of the `StorageHandledByL2` as specified in the EIP-5559.

## Related Issue

#15 

## Changes
- [X] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [ ] Other (please specify)

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -62,6 +62,38 @@
...
+      },
+      "console": "integratedTerminal",
+      "outFiles": [
+        "${workspaceFolder}/**/*.js",
+        "!**/node_modules/**"
+      ],
+      "args": [
+        "--resolver",
+        "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+        "-pk",
```
Comment (LeonardoVieira1630) [inline .vscode/launch.json:91]:
```suggestion
        "--privateKey",
```

File: packages/client/src/client.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+export async function handleL2Storage({
+  chainId,
+  l2Url,
+  args,
+}: {
+  chainId: bigint
+  l2Url: string
+  args: {
+    abi: Abi | unknown[]
```
Comment (LeonardoVieira1630) [inline packages/client/src/client.ts:57]: why does it needs to receive "unknown"? Some kind of patter that we are following?

File: packages/client/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  const l2Client = createPublicClient({
+    chain,
+    transport: http(l2Url),
+  }).extend(walletActions)
+
+  try {
+    const { request } = await l2Client.simulateContract(args)
+    await l2Client.writeContract(request)
+  } catch (err) {
+    console.log({ err })
```
Comment (LeonardoVieira1630) [inline packages/client/src/client.ts:75]:
```suggestion
    console.log("error while trying to make the request: ", { err })
```

File: packages/client/src/write.ts [hunk 4]
```diff
@@ -74,117 +62,76 @@ const _ = (async () => {
...
+        const [domain, url, message] = data.args as [
+          DomainData,
+          string,
+          MessageData,
+        ]
+        await handleDBStorage({ domain, url, message, signer })
+        break
+      }
+      default:
+        console.error({ err })
```
Comment (LeonardoVieira1630) [inline packages/client/src/write.ts:84]:
```suggestion
        console.error('writing failed: ', { err })
```

File: packages/client/src/write.ts [hunk 5]
```diff
@@ -74,117 +62,76 @@ const _ = (async () => {
...
+        handleL2Storage({
+          chainId,
+          l2Url: providerL2,
+          args: {
+            functionName: 'setText',
+            abi: l2Abi,
+            args: [
+              namehash(normalize(publicAddress)),
+              'com.twitter',
+              '@tartaruga.eth',
```
Comment (LeonardoVieira1630) [inline packages/client/src/write.ts:124]:
```suggestion
              '@blockful.eth',
```

Comment (LeonardoVieira1630) [reply packages/client/src/write.ts:124]: 🐢

File: .vscode/launch.json (hunk 1 above)
Comment (pikonha) [reply .vscode/launch.json:91]: there's a shorthand for this flag: https://github.com/blockful-io/external-resolver/pull/103/files#diff-fddecc9ce4eede7f9349805acc9070a43de231550d2b8c68a0c1ec32bac01e19R33

File: packages/client/src/client.ts (hunk 2 above)
Comment (pikonha) [reply packages/client/src/client.ts:57]: It is a pattern used by Viem: https://github.com/wevm/viem/blob/main/src/clients/decorators/public.ts#L324

**Pull request merged**